File: app/src/main/java/com/leeeyou/zhihudaily/view/ZhiHuDailyDetailActivity.java
package com.leeeyou.zhihudaily.view;

import android.os.Bundle;
import android.webkit.WebView;

import com.leeeyou.BaseActivity;
import com.leeeyou.R;
import com.leeeyou.util.HtmlUtils;
import com.leeeyou.zhihudaily.model.ZhiHuDailyRepositoryKt;
import com.leeeyou.zhihudaily.model.bean.ZhiHuDailyDetail;

import rx.Subscriber;
import rx.android.schedulers.AndroidSchedulers;
import rx.functions.Action0;
import rx.schedulers.Schedulers;

public class ZhiHuDailyDetailActivity extends BaseActivity {

    private int storyId;

    WebView story_web;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_story_detail);

        story_web = findViewById(R.id.story_web);

        storyId = getIntent().getIntExtra("storyId", -1);
        String storyTitle = getIntent().getStringExtra("storyTitle");

        setLeftTitleAndDisplayHomeAsUp(storyTitle);
        initWebView();
        getStoryDetail();
    }

    private void initWebView() {
        story_web.setVerticalScrollBarEnabled(false);
        story_web.getSettings().setDefaultTextEncodingName("UTF-8");
    }


    public void getStoryDetail() {
        ZhiHuDailyRepositoryKt
                .fetchZhiHuDailyDetailById(storyId)
                .subscribeOn(Schedulers.newThread())
                .doOnSubscribe(new Action0() {
                    @Override
                    public void call() {
//                        mMaterialDialog = new MaterialDialog.Builder(StoryDetailActivity.this)
//                                .content("Loading ...")
//                                .progress(true, 0)
//                                .show();
                    }
                })
                .subscribeOn(AndroidSchedulers.mainThread())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Subscriber<ZhiHuDailyDetail>() {
                    @Override
                    public void onCompleted() {
//                        mMaterialDialog.dismiss();
                    }

                    @Override
                    public void onError(Throwable e) {
//                        mMaterialDialog.dismiss();
                        e.printStackTrace();
                    }

                    @Override
                    public void onNext(ZhiHuDailyDetail storyDetail) {
                        story_web.loadData(HtmlUtils.structHtml(storyDetail.getBody(), storyDetail.getCss()), "text/html; charset=UTF-8", null);
                    }
                });
    }

}

File: app/src/main/java/com/leeeyou/zhihudaily/view/ZhiHuDailyDetailActivity.kt
package com.leeeyou.zhihudaily.view

import android.content.Intent
import android.os.Bundle
import com.afollestad.materialdialogs.MaterialDialog
import com.leeeyou.R
import com.leeeyou.manager.BaseActivity
import com.leeeyou.manager.BrowserActivity
import com.leeeyou.manager.PARAM_MODE
import com.leeeyou.manager.PARAM_URL
import com.leeeyou.manager.sonic.SonicJavaScriptInterface
import com.leeeyou.util.HtmlUtils
import com.leeeyou.zhihudaily.model.bean.ZhiHuDailyDetail
import com.leeeyou.zhihudaily.model.fetchZhiHuDailyDetailById
import kotlinx.android.synthetic.main.activity_story_detail.*
import rx.Subscriber
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers


class ZhiHuDailyDetailActivity : BaseActivity() {
    internal lateinit var mMaterialDialog: MaterialDialog

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_story_detail)
        setLeftTitleAndDisplayHomeAsUp(intent.getStringExtra("storyTitle"))
        initWebView()
        fetchZhiHuDetail()
    }

    private fun fetchZhiHuDetail() {
        fetchZhiHuDailyDetailById(intent.getIntExtra("storyId", -1))
                .subscribeOn(Schedulers.newThread())
                .doOnSubscribe {
                    mMaterialDialog = MaterialDialog.Builder(this@ZhiHuDailyDetailActivity)
                            .content("Loading...")
                            .progress(true, 0)
                            .show()
                }
                .subscribeOn(AndroidSchedulers.mainThread())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(object : Subscriber<ZhiHuDailyDetail>() {
                    override fun onCompleted() {
                        mMaterialDialog.dismiss()
                    }

                    override fun onError(e: Throwable) {
                        mMaterialDialog.dismiss()
                        e.printStackTrace()
                    }

                    override fun onNext(storyDetail: ZhiHuDailyDetail) {
                        zhiHuDailyWebview.loadData(HtmlUtils.structHtml(storyDetail.body, storyDetail.css), "text/html; charset=UTF-8", null)
//                        startBrowserActivity(MODE_SONIC, storyDetail.share_url)
                    }
                })
    }

    private fun startBrowserActivity(mode: Int, url: String) {
        val intent = Intent(this, BrowserActivity::class.java)
        intent.putExtra(PARAM_URL, url)
        intent.putExtra(PARAM_MODE, mode)
        intent.putExtra(SonicJavaScriptInterface.PARAM_CLICK_TIME, System.currentTimeMillis())
        startActivity(intent)
    }

    private fun initWebView() {
        zhiHuDailyWebview.isVerticalScrollBarEnabled = false
        zhiHuDailyWebview.settings.defaultTextEncodingName = "UTF-8"
    }

}